services/kvstore/src/main/java/org/hyperledger/besu/services/kvstore/LimitedInMemoryKeyValueStorage.java

```diff
@@ -59,7 +62,13 @@ public void clear() {
 
   @Override
   public boolean containsKey(final byte[] key) throws StorageException {
-    return get(key).isPresent();
+    final Lock lock = rwLock.readLock();
+    lock.lock();
+    try {
+      return get(key).isPresent();
```

Comment:
`get()` already does the locking internally - I don't think you need this extra locking